runtime/src/asset.rs

```diff
@@ -1962,4 +2035,96 @@ impl<T: Trait> Module<T> {
 
         Ok(())
     }
+
+    pub fn verify_restriction(
+        ticker: &Ticker,
+        extension_caller: T::AccountId,
+        from_did: Option<IdentityId>,
+        to_did: Option<IdentityId>,
+        value: T::Balance,
+        dest: T::AccountId,
+    ) -> RestrictionResult {
+        // 4 byte selector of verify_transfer - 0xD9386E41
+        let selector = hex!("D9386E41");
+        let balance_to = match to_did {
+            Some(did) => T::Balance::encode(&<BalanceOf<T>>::get((ticker, &did))),
+            None => T::Balance::encode(&(0.into())),
+        };
+        let balance_from = match from_did {
+            Some(did) => T::Balance::encode(&<BalanceOf<T>>::get((ticker, &did))),
+            None => T::Balance::encode(&(0.into())),
+        };
+        let encoded_to = Option::<IdentityId>::encode(&to_did);
+        let encoded_from = Option::<IdentityId>::encode(&from_did);
+        let encoded_value = T::Balance::encode(&value);
+        let total_supply = T::Balance::encode(&<Tokens<T>>::get(&ticker).total_supply);
+
+        // Creation of the encoded data for the verifyTransfer function of the extension
+        // i.e fn verify_transfer(
+        //        from: IdentityId,
+        //        to: IdentityId,
+        //        value: Balance,
+        //        balance_from: Balance,
+        //        balance_to: Balance,
+        //        total_supply: Balance
+        //    ) -> RestrictionResult { }
+
+        let encoded_data = [
+            &selector[..],
+            // &encoded_from[..],
+            // &encoded_to[..],
```

Comment:
the encoded data must always pass all values because it can't tell if its calling ptm or any other extension.